src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/RuntimeMethodHandleNode.cs

```diff
@@ -73,20 +74,29 @@ protected override ObjectData GetDehydratableData(NodeFactory factory, bool relo
             objData.RequireInitialPointerAlignment();
             objData.AddSymbol(this);
 
-            int handle = relocsOnly ? 0 : factory.MetadataManager.GetMetadataHandleForMethod(factory, _targetMethod.GetTypicalMethodDefinition());
+            int flags = 0;
+
+            MethodDesc targetMethodForMetadata = _targetMethod.GetTypicalMethodDefinition();
+            if (targetMethodForMetadata.IsAsyncVariant())
+            {
+                targetMethodForMetadata = factory.TypeSystemContext.GetTargetOfAsyncVariantMethod(targetMethodForMetadata);
+                flags |= RuntimeMethodHandleConstants.IsAsyncVariant;
+            }
+
+            int handle = relocsOnly ? 0 : factory.MetadataManager.GetMetadataHandleForMethod(factory, targetMethodForMetadata);
 
             objData.EmitPointerReloc(factory.MaximallyConstructableType(_targetMethod.OwningType));
             objData.EmitInt(handle);
 
             if (_targetMethod != _targetMethod.GetMethodDefinition())
             {
-                objData.EmitInt(_targetMethod.Instantiation.Length);
+                objData.EmitInt(_targetMethod.Instantiation.Length | flags);
```

Comment:
We will use 5 bytes to encode async variants. Hopefully, the async variant methods are rare enough for it to matter.